src/java_bytecode/java_bytecode_typecheck_expr.cpp

```diff
@@ -254,7 +275,7 @@ void java_bytecode_typecheckt::typecheck_expr_symbol(symbol_exprt &expr)
   else
   {
     // yes!
-    assert(!s_it->second.is_type);
+    PRECONDITION(!s_it->second.is_type);
```

Comment:
I think that's an `INVARIANT` (it's not a precondition on our input, but a state invariant)